lib/resolvers.js

```diff
@@ -29,3 +30,17 @@ export const mapLinkList = fieldName => doc => {
     return linkItem;
   }).filter(l => l !== undefined);
 };
+
+export const mapScheduleList = fieldName => doc => {
+  const schedule = doc[fieldName];
+
+  if (!schedule) { return []; }
+  return schedule.map((item) => {
```

Comment:
@jameshopkins If we're doing it for the sake of efficiency I would expect Chris' to actually be less efficient then the original one as it's reallocating the array on each iteration.

Immutable enum building like this is only efficient with singly linked lists.